tests/test_real_services.py

```diff
@@ -105,27 +105,30 @@ def test_nominative_fetch_data_cases(street, house_number, mock_type, mock_data,
 @respx.mock
 @pytest.mark.parametrize("coordinate, radius, mock_type, mock_data, expected", [
     # Happy path - returns features list
-    ((34.7735910, 32.0698820), 100, "success", gisn_resp, gisn_resp["features"]),
+    ((34.7735910, 32.0698820), 100, "success", gisn_resp,
+     gisn_resp["features"]),
 
     # Empty features list
     ((34.7735910, 32.0698820), 100, "success", {"features": []}, []),
 
-    # HTTP 404 error
-    ((34.7735910, 32.0698820), 100, "http_error", {"status_code": 404, "text": "Not Found"}, (404, b'{"error": "GISN API error: 404 Not Found"}')),
+    # HTTP 404 error - should raise exception
+    ((34.7735910, 32.0698820), 100, "http_error",
+     {"status_code": 404, "text": "Not Found"}, "exception"),
 
-    # HTTP 500 error
-    ((34.7735910, 32.0698820), 100, "http_error", {"status_code": 500, "text": "Internal Server Error"}, (500, b'{"error": "GISN API error: 500 Internal Server Error"}')),
+    # HTTP 500 error - should raise exception
+    ((34.7735910, 32.0698820), 100, "http_error",
+     {"status_code": 500, "text": "Internal Server Error"}, "exception"),
 
-    # Network timeout/request error - NEW TEST CASE
-    ((34.7735910, 32.0698820), 100, "request_error", None, (503, b'{"error": "GISN API request failed:  Timeout"}')),
+    # Network timeout/request error - should raise exception
+    ((34.7735910, 32.0698820), 100, "request_error", None, "exception"),
 ])
 def test_gisn_fetch_data_cases(coordinate, radius, mock_type, mock_data, expected):
     if mock_type == "success":
```

Comment:
You usually don't want these kind of if-that-then-this in your test code - it's better to write a couple of separate tests each handling specific situation (success, failure, exceptions, etc.)